frontend/src/metabase/nav/components/SearchBar.tsx

```diff
@@ -166,18 +194,32 @@ function SearchBarView({ location, onSearchActive, onSearchInactive }: Props) {
           onKeyPress={handleInputKeyPress}
           ref={searchInput}
         />
+        <SearchFunnelButton
+          icon="filter"
+          data-testid={
+            isFiltered
```

Comment:
We rarely do conditional `data-testid`, I think I found less than a couple of places where we set `data-testid` conditinally.

Can we use a static name? Because using a different name doesn't guarantee this search bar will be highlighted. since it's targeting the same element. I think we should assert highlighted search bar a different way that gives us more confidence.

Reply:
What would you propose? I was thinking that we might be able to use another data attribute or something, but I haven't seen too many examples that aren't asserting on a classname (and we're also trying to move away from that)

Reply:
Asserting class name would be bad. I think asserting data attributes sounds a lot better. e.g. `data-is-highlighted={isFiltered}`

Reply:
lit, I'll add that

Reply:
Solved with
```
 expect(screen.getByTestId("search-bar-filter-button")).toHaveAttribute(
        "data-is-filtered",
        <"true" | "false">,
      );
```